qiskit/pulse/parser.py

```diff
@@ -182,10 +182,23 @@ def visit_Num(self, node: ast.Num) -> ast.Num:
         Returns:
             Input node.
         """
+        # node that Num node is deprecated in Python 3.8.
+        # Constant node is recommended.
         return node
 
-    def visit_Name(self, node: ast.Name) -> Union[ast.Name, ast.Num]:
-        """Evaluate name and return ast.Num if it is bound.
+    def visit_Constant(self, node: ast.Constant) -> ast.Constant:
```

Comment:
Just for my own understanding, where did this come up?

Reply:
Please check this https://docs.python.org/3/library/ast.html#ast.AST

> Deprecated since version 3.8: Old classes ast.Num, ast.Str, ast.Bytes, ast.NameConstant and ast.Ellipsis are still available, but they will be removed in future Python releases. In the meantime, instantiating them will return an instance of a different class.

Now constructor of `ast.Num` returns `ast.Constant` and the parser checks if all parameters are bound by calling `isinstance` (previously ast node becomes `ast.Num` is expression is evaluated but now `ast.Constant` and this check caused bug).